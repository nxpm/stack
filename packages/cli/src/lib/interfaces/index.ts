export type WebStyleLibrary = 'bootstrap' | 'tailwind'
