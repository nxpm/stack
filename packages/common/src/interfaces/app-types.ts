export enum AppTypeApi {
  nest = 'nest',
}

export enum AppTypeMobile {
  ionicAngular = 'ionic-angular',
}

export enum AppTypeWeb {
  angular = 'angular',
}
