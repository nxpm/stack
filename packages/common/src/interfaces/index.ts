export * from './app-types'
export * from './base-schema'
export * from './normalized-schema'
