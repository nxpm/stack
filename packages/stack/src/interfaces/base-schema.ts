export interface BaseSchema {
  apiName?: string
  mobileName?: string
  webName?: string
  name?: string
  tags?: string
  directory?: string
}
