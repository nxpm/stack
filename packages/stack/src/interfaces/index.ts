export * from './base-schema'
export * from './normalized-schema'
