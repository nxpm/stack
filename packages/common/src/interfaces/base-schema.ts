import { AppTypeApi, AppTypeMobile, AppTypeWeb } from './app-types'

export interface BaseSchema {
  apiName?: string
  apiType?: AppTypeApi
  mobileName?: string
  mobileType?: AppTypeMobile
  webName?: string
  webType?: AppTypeWeb
  skipApi?: boolean
  skipMobile?: boolean
  skipWeb?: boolean
  name: string
  tags?: string
  directory?: string
}
