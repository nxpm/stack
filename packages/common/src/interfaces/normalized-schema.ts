import { BaseSchema } from './base-schema'
import { AppTypeApi, AppTypeMobile, AppTypeWeb } from './app-types'

export interface NormalizedSchema extends BaseSchema {
  npmScope: string
  appNameApi: string
  appNameMobile: string
  appNameWeb: string
  appTypeApi: AppTypeApi
  appTypeMobile: AppTypeMobile
  appTypeWeb: AppTypeWeb
  skipApi: boolean
  skipMobile: boolean
  skipWeb: boolean
  projectName?: string
  projectRoot?: string
  projectDirectory?: string
  parsedTags?: string[]
}
