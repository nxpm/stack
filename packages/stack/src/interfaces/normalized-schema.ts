import { BaseSchema } from './base-schema'

export interface NormalizedSchema extends BaseSchema {
  npmScope: string
  appNameApi: string
  appNameMobile: string
  appNameWeb: string
  projectName?: string
  projectRoot?: string
  projectDirectory?: string
  parsedTags?: string[]
}
