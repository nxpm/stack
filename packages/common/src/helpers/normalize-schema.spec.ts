import { Tree } from '@nrwl/devkit'
import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing'
import { BaseSchema } from '../interfaces'
import { normalizeOptions } from '../helpers/normalize-schema.helper'

describe('normalize-schema', () => {
  let appTree: Tree
  const options: BaseSchema = { name: 'test' }

  beforeEach(() => {
    appTree = createTreeWithEmptyWorkspace()
  })

  it('should run successfully', async () => {
    const schema = normalizeOptions(appTree, options)
    expect(schema).toMatchSnapshot()
  })
})
