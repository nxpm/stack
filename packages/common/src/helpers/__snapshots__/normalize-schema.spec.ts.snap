// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`normalize-schema should run successfully 1`] = `
Object {
  "appNameApi": "api",
  "appNameMobile": "mobile",
  "appNameWeb": "web",
  "appTypeApi": "nest",
  "appTypeMobile": "ionic-angular",
  "appTypeWeb": "angular",
  "name": "test",
  "npmScope": "proj",
  "skipApi": false,
  "skipMobile": false,
  "skipWeb": false,
}
`;
